email/src/main/java/io/micronaut/email/Attachment.java

```diff
@@ -0,0 +1,176 @@
+ * limitations under the License.
+ */
+package io.micronaut.email;
+
+import io.micronaut.core.annotation.Introspected;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.util.Objects;
+
+/**
+ * Email's Attachment.
+ * @author Sergio del Amo
+ * @since 1.0.0
+ */
+@Introspected
+public class Attachment {
+
+    @NonNull
+    @NotBlank
+    private final String filename;
+
+    @NonNull
+    @NotNull
+    private final byte[] content;
+
+    @NonNull
+    @NotBlank
+    private final String contentType;
+
+    @Nullable
+    private final String id;
+
+    /**
+     *
+     * @param filename filename to show up in email
+     * @param content file content
+     * @param contentType file content type
+     * @param id content identifier
+     */
+    public Attachment(@NonNull String filename,
+                      @NonNull String contentType,
+                      @NonNull byte[] content,
+                      @Nullable String id) {
+        this.filename = filename;
+        this.content = content;
+        this.contentType = contentType;
+        this.id = id;
+    }
+
+    /**
+     *
+     * @return Attachment's builder
+     */
+    @NonNull
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     *
+     * @return filename to show up in email
+     */
+    @NonNull
+    public String getFilename() {
+        return this.filename;
+    }
+
+    /**
+     *
+     * @return file content type
+     */
+    @NonNull
+    public String getContentType() {
+        return this.contentType;
+    }
+
+    /**
+     *
+     * @return file content type
+     */
+    @NonNull
+    public byte[] getContent() {
+        return this.content;
+    }
+
+    /**
+     *
+     * @return Content Id
+     */
+    @Nullable
+    public String getId() {
+        return this.id;
+    }
+
+    /**
+     * Attachment's builder.
+     */
+    public static class Builder {
+        private String filename;
+        private byte[] content;
+        private String contentType;
+        private String id;
+
+        /**
+         *
+         * @param filename filename to show up in email
+         * @return Attachment's builder
+         */
+        @NonNull
+        public Builder filename(@NonNull String filename) {
+            this.filename = filename;
+            return this;
+        }
+
+        /**
+         *
+         * @param contentType file content type
+         * @return Attachment's builder
+         */
+        @NonNull
+        public Builder contentType(@NonNull String contentType) {
+            this.contentType = contentType;
+            return this;
+        }
+
+        /**
+         *
+         * @param content file content
+         * @return Attachment's builder
+         */
+        @NonNull
+        public Builder content(@NonNull byte[] content) {
```

Comment:
it is convenient to add more options to specify content - at least `InputStream` and `File`. there is plenty of responses on SFO how to convert `File` or `InputStream` to `byte[]` but why forcing the developers to Google stuff when you can provide them a solution right ahead.